media/Circletracer/treemap.js

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+let circleJson, durCircleJson;
+function treeMap(json) {
+  circleJson = json;
+  durCircleJson = json;
+  let g = new dagreD3.graphlib.Graph().setGraph({});
+  let nodes = [];
+  let inputNode, outputNode;
+
+  g.setNode(0, {label: 'input', class: 'type-input'});
+  json.forEach((element, idx) => {
+    let type = element.properties.type;
+    let myIndex = element.outputs[0].location;
+    let parentsIndex = [];
+    let location = element.properties.location;
+    let attributes = element.attributes;
+    let inputs = element.inputs;
+    let outputs = element.outputs;
+    inputs.forEach(input => {
+      if (input.edge === true) {
+        parentsIndex.push(input);
+      }
+    });
+
+    let node = {
+      'index': myIndex,
+      'type': type,
+      'class': 'type-' + type,
+      'parents': parentsIndex,
+      'location': location,
+      'attributes': attributes,
+      'inputs': inputs,
+      'outputs': outputs
+    };
+
+    let label = `<p class='type'>${type}</p>`;
+
+    // inputs label create
+    inputs.forEach((input, checkIdx) => {
+      if (checkIdx === 0 || input.edge === true) {
+        return;
+      }
+
+      label += `<p><label><b>input${checkIdx}</b></label><span>&lt;${
+          getTypeArray('x', input.type)}&gt;</span></p>`;
+    });
+
+    attributes.forEach(attr => {
+      if (attr['attribute'] === 'fused_activation_function' && attr['value'] !== 'NONE') {
+        label += `<p class='activation'>${attr['value']}</p>`;
+      }
+    });
+
+    // if element has duration
+    if (element.hasOwnProperty('duration')) {
+      let timeUnit = element.duration.timeUnit;
+      let dur1 = element.duration.dur1;
+      let dur2 = element.duration.dur2;
+
+      label += `<p class='duration-title'>DURATION</p>`;
+      label += `<p class='duration'>${dur1.toFixed(4)} ${timeUnit} <b>&roarr;</b> ${
+          dur2.toFixed(4)} ${timeUnit}</p>`;
+    }
+
+    // First node logic
```

Comment:
Hmm... there can be multiple input nodes and multiple output nodes...?

Reply:
If you are talking about lines 78 to 106, this is the code that handles the first and last nodes. Therefore, it is a code that does not work when it is not the first node and the last node, and it is a necessary code in the code :)

Reply:
But according to the variable name, I think you are treating first node as graph input and last node as graph output.
Am I understanding correctly?

Reply:
Yes. I think I answered wrong. In the current logic, it is correct that the input node and output node are one. If there are multiple cases, it seems that correction is necessary.